### PR TITLE
Make 404 handler prettier

### PR DESCRIPTION
**Pull request by ithinkihaveacat**
Tidy up the 404 handler. (Need something similar for #69.)

### CODE REVIEW

File: controllers/pwas/crud.js [hunk 1]
```diff
@@ -136,7 +136,8 @@ router.post('/:pwa/edit', (req, res, next) => {
 router.get('/:pwa', (req, res, next) => {
   pwaModel.find(req.params.pwa, (err, entity) => {
     if (err) {
-      return next(err);
+      // Not really an error: the pwa wasn't found in the db. Fall through to 404 page.
```
Comment (andreban) [inline controllers/pwas/crud.js]: Should the find method invoke the callback(null, null), so both err and entity are null? Maybe it's more clear in the code than testing an error that's not exactly an error.

**Comment by andreban** [comment]
LGTM, with a comment that we can deal with later.

**Comment by juliantoledo** [comment]
LGTM.

Let's also create another PR with tests for all the entry points, including this 404s.

File: controllers/pwas/crud.js (hunk 1 above)
Comment (ithinkihaveacat) [inline controllers/pwas/crud.js]: Strangely this error comes all the way from https://github.com/GoogleChrome/gulliver/blob/master/lib/model-datastore.js#L175. It appears `get()` does not distinguish between a "Bad Request" and a key not found.

Not sure of the "right" way to fix this--pre-flight the `get()` with a query for the key? But that would be two trips. @juliantoledo do you know?

Anyway, I don't think this is a problem for this PR, will punt for later.

Comment (juliantoledo) [inline controllers/pwas/crud.js]: We should handle them differently I think, let's do a PR for error handling fixes.

Comment (ithinkihaveacat) [inline controllers/pwas/crud.js]: @juliantoledo See #81.

**Comment by ithinkihaveacat** [comment]
@juliantoledo See #80 for the tests. You want to add a milestone?

**Pull request merged**